src/core/raster/qgsrasterattributetable.h

```diff
@@ -262,24 +307,116 @@ class CORE_EXPORT QgsRasterAttributeTable
+     * if there is not match.
+     */
+    QVariantList row( const double matchValue ) const;
+
+    /**
+     * Returns the classes for a thematic Raster Attribute Table, classified
+     * by \a classificationColumn, the default value of -1 makes the method guess
+     * the classification column based on the field usage.
+     */
+    QList<QgsRasterAttributeTable::MinMaxClass> minMaxClasses( const int classificationColumn  = -1 ) const;
+
+    /**
+     * Returns the color ramp for an athematic Raster Attribute Table
+     * setting the labels in \a labels, optionally generated from \a labelColumn.
+     */
+    QgsGradientColorRamp colorRamp( QStringList &labels SIP_OUT, const int labelColumn = -1 ) const;
+
+    /**
+     * Creates and returns a (possibly NULLPTR) raster renderer for the
+     * specified \a provider and \a bandNumber and optionally reclassified
+     * by \a classificationColumn, the default value of -1 makes the method
+     * guess the classification column based on the field usage.
+     *
+     * \note athematic attribute tables with color ramps cannot be reclassified,
+     *       the renderer will still use the \a classificationColumn for
+     *       generating the class labels.
+     */
+    QgsRasterRenderer *createRenderer( QgsRasterDataProvider *provider, const int bandNumber, const int classificationColumn = -1 ) SIP_FACTORY;
+
+    /**
+     * Returns the data rows ordered by the value column(s) in ascending order, if
+     * the attribute table type is athematic the middle value for each row range
+     * is considered for ordering.
+     * If the attribute table does not have any value field (and hence is not valid),
+     * the current data are returned without any change.
+     */
+    QList<QList<QVariant>> orderedRows( ) const;
 
     /**
      * Try to determine the field usage from its \a name and \a type.
      */
     static Qgis::RasterAttributeTableFieldUsage guessFieldUsage( const QString &name, const QVariant::Type type );
 
+    /**
+    * Returns the (possibly empty) path of the file-based RAT, the path is set when a RAT is read or written from/to a file.
+    *
+    * \see writeToFile()
+    * \see readFromFile()
+    */
+    const QString &filePath() const;
+
     /**
      * Returns the translated human readable name of \a fieldUsage.
+     * \see usageInformation()
      */
     static QString usageName( const Qgis::RasterAttributeTableFieldUsage fieldusage );
 
+    /**
+     * Returns the list of field usages for colors and values.
+     */
+    static QList<Qgis::RasterAttributeTableFieldUsage> valueAndColorFieldUsages();
+
+    /**
+     * Creates a new Raster Attribute Table from a \a rasterLayer, the renderer must be Paletted or SingleBandPseudoColor, optionally reporting in \a bandNumber the raster band from which the attribute table was created.
+     * \returns NULL in case of errors or unsupported renderer.
```

Comment:
if you use SIP_OUT annotation then you should explicitly include all "\param ..." values and a "\returns" value, so that sipify can do its magic correctly